tools/updater/main.c

```diff
@@ -934,174 +862,220 @@ static k4a_result_t command_inspect_firmware(updater_command_info_t *command_inf
 
 static k4a_result_t command_update_device(updater_command_info_t *command_info)
 {
-    firmware_status_summary_t finalStatus;
-
-    k4a_result_t result = ensure_firmware_open(command_info);
-    if (!K4A_SUCCEEDED(result))
-    {
-        return result;
-    }
+    firmware_status_summary_t finalFwCommandStatus;
+    k4a_result_t finalCmdStatus = K4A_RESULT_SUCCEEDED;
 
-    // Query the current device information...
-    result = command_query_device(command_info);
-    if (!K4A_SUCCEEDED(result))
+    for (uint32_t device_index = 0; device_index < command_info->device_count; device_index++)
     {
-        return result;
-    }
+        k4a_result_t result = ensure_firmware_open(command_info, device_index);
+        if (!K4A_SUCCEEDED(result))
+        {
+            return result;
+        }
 
-    // Load and parse the firmware file information...
-    result = command_inspect_firmware(command_info);
-    if (!K4A_SUCCEEDED(result))
-    {
-        return result;
-    }
+        // Make a copy of the command so that only 1 device is updated at a time.
+        updater_command_info_t sub_command_info = { 0 };
+        sub_command_info.device_count = 1;
+        sub_command_info.device_serial_number[0] = command_info->device_serial_number[device_index];
+        sub_command_info.firmware_handle = command_info->firmware_handle;
+        sub_command_info.firmware_serial_number = command_info->firmware_serial_number;
```

Comment:
I might clear command_info->firmware_handle and firmware_serial_number as sub_command_info now own both.